File: ui/src/app/model/OrgAdminDashboardVO.ts
import { IncidentModuleVO } from "./IncidentModuleVO";
import { IncidentStatusVO } from "./IncidentStatusVO";

export class OrgAdminDashboardVO {
    activeEmployees: number;
    inHouseVendors: number;
    outSourcedVendors: number;
    moduleWiseIncidents: IncidentModuleVO[];
    statusWiseIncidents: IncidentStatusVO[];
    moduleWiseAssetIncidents: IncidentModuleVO[];
    statusWiseAssetIncidents: IncidentStatusVO[];

    constructor(data: any) {
        if (data == null) {
            this.activeEmployees = -1;
            this.inHouseVendors = -1;
            this.outSourcedVendors = -1;
            this.moduleWiseIncidents = [];
            this.statusWiseIncidents = [];
            this.moduleWiseAssetIncidents = [];
            this.statusWiseAssetIncidents = [];
        } else {
            this.activeEmployees = data.activeEmployees;
            this.inHouseVendors = data.inHouseVendors;
            this.outSourcedVendors = data.outSourcedVendors;
            this.moduleWiseIncidents = data.moduleWiseIncidents;
            this.statusWiseIncidents = data.statusWiseIncidents;
            this.moduleWiseAssetIncidents = data.moduleWiseAssetIncidents;
            this.statusWiseAssetIncidents = data.statusWiseAssetIncidents;
        }
    }
}

File: domain/src/main/java/com/centram/common/vo/OrgAdminDashboardVO.java
package com.centram.common.vo;


import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode
public class OrgAdminDashboardVO {
    private long activeEmployees;
    private long inHouseVendors;
    private long outSourcedVendors;
    private Set<IncidentModuleVO> moduleWiseIncidents;
    private Set<IncidentStatusVO> statusWiseIncidents;

    public OrgAdminDashboardVO(long activeEmployees) {
        this.activeEmployees = activeEmployees;
    }

    public OrgAdminDashboardVO(long inHouseVendors, long outSourcedVendors) {
        this.inHouseVendors = inHouseVendors;
        this.outSourcedVendors = outSourcedVendors;
    }

    public OrgAdminDashboardVO(long activeEmployees, long inHouseVendors, long outSourcedVendors) {
        this.activeEmployees = activeEmployees;
        this.inHouseVendors = inHouseVendors;
        this.outSourcedVendors = outSourcedVendors;
    }
}